### PR TITLE
Fix cache path

### PR DESCRIPTION
**Pull request by KevinRansom**
There appears to be circumstances where Path.Combine treats elements not at the end of the list as extensions.

This issue here:  https://github.com/dotnet/fsharp/issues/15294
contains this error:
````
#r "nuget:NBomber"
^^^^^^^^^^^^^^^^^^
stdin(1,1): error FS3217: C:\Users\woo.packagemanagement\nuget\Projects\1220--24e02601-e93e-4b40-a9bf-33a26d273aa0\Project.fsproj : error NU1101: 找不到包 NBomber。源 C:\Program Files\dotnet\library-packs, C:\Program Files\dotnet\sdk\8.0.100-preview.4.23260.5\FSharp\library-packs 中不存在具有此 ID 的包
````
The actual error is unrelated to this, however, the project was created in the wrong location:
````
C:\Users\woo.packagemanagement\nuget\Projects\1220--24e02601-e93e-4b40-a9bf-33a26d273aa0\Project.fsproj 
````
It should have been:
````
C:\Users\woo\.packagemanagement\nuget\Projects\1220--24e02601-e93e-4b40-a9bf-33a26d273aa0\Project.fsproj 
````
It seems as if there are circumstance where:
````
        let specialDir =
            Path.Combine(Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile), ".packagemanagement", "nuget")
````
combines the .packagemanagement as a file extension.  I haven't been able to reproduce this locally.  Perhaps it is locale related.  This pr constructs the path more specifically

### CODE REVIEW

File: src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs [hunk 1]
```diff
@@ -312,7 +312,19 @@ type FSharpDependencyManager(outputDirectory: string option, useResultsCache: bo
...
-            Path.Combine(Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile), ".packagemanagement", "nuget")
+            let getProfilePath =
+                // If it has a directory seperator remove it
+                let path = Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile)
+
+                if
+                    (path.EndsWith(Path.DirectorySeparatorChar.ToString()))
+                    && not (path.EndsWith(Path.AltDirectorySeparatorChar.ToString()))
+                then
+                    path.Substring(0, path.Length)
```
Comment (T-Gro) [inline src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs:323]:
```suggestion
                    path.Substring(0, path.Length - 1)
```

File: src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs [hunk 2]
```diff
@@ -312,7 +312,19 @@ type FSharpDependencyManager(outputDirectory: string option, useResultsCache: bo
...
         //   if the path wasn't supplied or not rooted use the temp directory as the root.
         let specialDir =
-            Path.Combine(Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile), ".packagemanagement", "nuget")
+            let getProfilePath =
+                // If it has a directory seperator remove it
+                let path = Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile)
+
+                if
+                    (path.EndsWith(Path.DirectorySeparatorChar.ToString()))
+                    && not (path.EndsWith(Path.AltDirectorySeparatorChar.ToString()))
```
Comment (T-Gro) [inline src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs:321]:
```suggestion
                    ||  (path.EndsWith(Path.AltDirectorySeparatorChar.ToString()))
```

Why DirectorySeparatorChar && NOT AltDirectorySeparatorChar ?
I would say we should not duplicate the separator if it is either of the two.

**Review by psfinaki — COMMENTED** [review]
Probably would be nice to write/generate some unit tests for such things.

**Comment by KevinRansom** [comment]
> Probably would be nice to write/generate some unit tests for such things.

@psfinaki --- the reported issue, isn't an issue we have any control over.  the NuGet.config file is not one we ever open, read , edit or anything else.  It is a user file containing configuration specified by various applications.  Indeed, it not having access to nuget.org is a normal, nuget scenario, and one in which we work fine.  The error message even says what's wrong.  It is certain that the corruption is not because of anything we do.

**Review by T-Gro — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]

**Pull request merged**